File: timestamp.py
from datetime import datetime
import json
import os

# set path to last_run.json
working_dir = os.path.dirname(__file__)
path = f'{working_dir}\\last_run.json'

''' check to see if last_run.json exists
    if it does exist print the last time
    the script was run, if it doesn't exist
    create the file'''
if os.path.exists(path):
    with open(path, 'r') as file:
        data = json.load(file)
        print(f'Last run on {data["last_run"]}')
else:
    open(path, 'w').close()

# get date and time
date = str(datetime.now())
month = [0]
day = [0]
year = [0]
hour = [0]

def parseDate():
    if int(date[5:7]) < 10:
        month[0] = date[6:7]
    elif int(date[5:7]) >= 10:
        month[0] = date[5:7]
    if int(date[8:10]) < 10:
        day[0] = date[9:10]
    elif int(date[8:10]) >= 10:
        day[0] = date[8:10]
    year[0] = date[:4]
    if int(date[11:13]) < 10:
        hour[0] = date[12:13]
    elif int(date[11:13]) >= 10:
        hour[0] = date[11:13]

parseDate()

timestamp = f'{month[0]}/{day[0]}/{year[0]} {hour[0]}:{date[14:16]}'
timestamp_dict = {"last_run":timestamp}

# write date to last_run.json
with open(path, 'w') as file:
    json.dump(timestamp_dict, file)

File: timestamp.js
const fs = require('fs');
const PATH = `${__dirname}/last_run.json`;

function timestamp() {
    fs.readFile(PATH, 'utf-8', (err, data) => {
      if (err) {
	console.log('Please run again.');
	console.log(err);
};
      const last_run = JSON.parse(data);
      console.log(`Last run on ${last_run['last_run']}`);
    });

    currentDate = new Date();
    month = currentDate.getMonth() + 1;  // getMonth uses 0-11 format
    day = currentDate.getDate();
    year = currentDate.getFullYear();
    hour = currentDate.getHours();
    // getMinutes doesn't have a leading 0 on numbers 0-9
    minute = () => {
      if (currentDate.getMinutes() < 10) {
        return `0${currentDate.getMinutes()}`
      } else {
        return currentDate.getMinutes()
      }
    };

    // write date and time to last_run.json
    timestamp = `${month}/${day}/${year} ${hour}:${minute()}`;
    timestamp_obj = {
      'last_run': timestamp,
    };
    timestamp_str = JSON.stringify(timestamp_obj);
    fs.writeFile(PATH, timestamp_str, (err) => {
        if(err) console.log(err);;
    });
};

module.exports = { timestamp };
